### PR TITLE
Feature/chromatogram extractor

### PR DESCRIPTION
**Pull request by hroest**
fixes for chromatogram extractor

- [FIX] chromatogram extraction for the very first data point
- add tests

### CODE REVIEW

**Comment by hroest** [comment]
@hendrikweisser this changes some things in FFId, can you maybe have a look? Basically there was an issue when the m/z value was the very first value in the chromatogram, then it would not get extracted by the algorithm.

**Comment by hroest** [comment]
can we merge this?

File: src/openms/source/ANALYSIS/OPENSWATH/ChromatogramExtractorAlgorithm.cpp [hunk 1]
```diff
@@ -94,20 +95,36 @@ namespace OpenMS
       integrated_intensity += (*int_walker);
     }
 
-    // walk to the left until we go outside the window, then start walking to the right until we are outside the window
+    // (i) Walk to the left one step and then keep walking left until we go
+    // outside the window. Note for the first step to the left we have to
+    // check for the walker becoming zero.
```
Comment (timosachsenberg) [inline src/openms/source/ANALYSIS/OPENSWATH/ChromatogramExtractorAlgorithm.cpp]: its not becoming zero in this part so it might be a copy & paste problem

**Comment by timosachsenberg** [comment]
looks good from my side

**Comment by hroest** [comment]
@hendrikweisser I looked at a few features that are now different than before. In the FFID 1 output, there is only a single feature that changes significantly:

- LALDLVVR/3 changes intensity by 3% as the integration is now a bit longer 1851 seconds vs 1843 seconds

all other changes are minor changes (rounding errors in the scores etc). So I dont think this actually affects FFId output that much.

**Comment by hroest** [comment]
if there are no further comments, I will go ahead and merge this sometime soon

**Comment by hroest** [comment]
Here is a screenshot of the difference, again not really a relevant change in my view

screenshot

**Comment by timosachsenberg** [comment]
last change to veto @hendrikweisser ;)

**Comment by hendrikweisser** [comment]
Fine with me. (Sorry for the delay in replying.)

**Pull request merged**